### ISSUE TITLE
Single CUDA device multistream support

### ISSUE THREAD

**Issue by useongkim** [arrayfire/arrayfire#3397]
I would like to request a feature that enables setting no only for cuda device id but also cuda stream explictily for each cpu thread.

Description
===========

The hardwares I am working with typically have 2 or 4 cuda devices with 12 to 24 cpu threads. Only a single process uses an individual cuda device because of the exclusive process mode.

I am using Arrayfire for tensor operations in cuda devices. (I am using the unified backend. libaf.so) Workloads in a single stream does not make the full cuda core utilitization when they are executed sequentially. So I would like to enable multi cuda streams in a single device so that the multiple workloads can run concurrently on a cuda device. 

In short, my request is enabling multiple ArrayFire operations run on multiple cuda streams.

I am aware of the tickets such as #113 and #2378 related to this. The last update seems in 2021.

Is there any update on multiple cuda stream support by ArrayFire?

Thank you.

Useong Kim
